### PR TITLE
(#1580) Update Storybook Buttons

### LINKED ISSUE #1580: Update storybook structure for usa-button
### `usa-button` storybook structure

:file_folder: Button
- :file_folder: Default
  - Default Button (Primary, Secondary, Accent Cool, Accent Warm, Base)
  - Default Anchor (Primary, Secondary, Accent Cool, Accent Warm, Base)
- :file_folder: Modifiers
  - :file_folder: Size
    - :file_folder: Big
      - [Big Button](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-usa-button--big) (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse, Icon Left, Icon Right)
      - Big Anchor (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse, Icon Left Primary, Icon Right Primary)
    - :file_folder: Small
      - [Small Button](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-usa-button--nci-small) (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse, Icon Left Primary, Icon Right Primary)
      - Small Anchor (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse)
  - :file_folder: Full Width
    - [Full Width Button](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-usa-button--nci-full-width)  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse, Icon Left Primary, Icon Right Primary)
    - Full-width Anchor  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse, Icon Left Primary, Icon Right Primary)
  - :file_folder: Icon
    - :file_folder: Icon, left
      - [Icon, left Button](https://designsystem-dev.cancer.gov/develop/storybook/?path=/story/components-usa-button--icon-button)  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse)
      - Icon, left Anchor  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse)
    - :file_folder: Icon, right
      - Icon, right Button  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse)
      - Icon, right Anchor  (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline Primary, Outline Secondary, Outline Inverse)
- :file_folder: Variants
  - :file_folder: Outline
    - Outline Button (Outline Primary, Outline Secondary, Outline Inverse)
    - Outline Anchor (Outline Primary, Outline Secondary, Outline Inverse)
  - :file_folder: Unstyled
    - Unstyled (Default; Big; Small; Full Width; Icon, Left; Icon, Right)
- :file_folder: Test cases
  - Unsupported Outline Colors (Primary, Secondary, Accent Cool, Accent Warm, Base)

Other tasks:
* Clean up all of the regressions (redundant breakpoint scenarios) associated with this component
* Make sure scenarios still reflect all of the test cases that we have, but we do not need them as examples in the storybook structure

### Additional details / screenshot
- [Storybook structure wiki](https://github.com/NCIOCPL/ncids/wiki/Storybook-Structure)

### Related Tickets
* Parent ticket: #1224

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #1580

### CODE REVIEW

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Can you please move the "Outline Inverse" label outside of the dark background?

**Comment by bennettcc** [comment]
@bryanpizzillo this one looks like its ready for your re-review please

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
@sarinapadilla, @andyvanavery31  and @olitharp-nci - I remember discussing that we need to have tests for icons as Big and Small. We should have it, but IDC if the use cases are added to big/small, or if big/small is added to the icon left/right. Also we definitely need tests for icon when it is full width (left and right). (The reason we definitely need them is that we had missed this once before and it caused https://github.com/NCIOCPL/cgov-digital-platform/issues/4242)

Additionally, we probably do not need all the different breakpoints for the buttons, unless there is some specific thing in that variant/modifier. Or for example, just put breakpoints on the default if there is something special with button, but not on all the other size/outline modifiers, for example.

**Comment by andyvanavery31** [comment]
@bryanpizzillo @olitharp-nci @sarinapadilla I think that trying to do them all inside the Modifiers could get confusing, so how about we make a subfolder within the Test Cases folder that is something like "Modifier Combinations". Then inside of it we can hit all the permutations we need to test, and for each one we put all the themes (Primary, Secondary, Accent Cool, Accent Warm, Base, Outline, Outline Inverse):

- Full Width Icon Left Anchor
- Full Width Icon Right Anchor
- Full Width Icon Left Button
- Full Width Icon Right Button
- Big Icon Left Anchor
- Big Icon Right Anchor
- Big Icon Left Button
- Big Icon Right Button
- Small Icon Left Anchor
- Small Icon Right Anchor
- Small Icon Left Button
- Small Icon Right Button

I also agree about what you said with the regression images

**Comment by karlikpj** [comment]
Icon buttons need at least desktop / mobile as the icon should stick to the  left or right side as the button snaps goes to full width at mobile. otherwise the rest can just be the default viewport size.

**Comment by bryanpizzillo** [comment]
> * Full Width Icon Left Anchor
> * Full Width Icon Right Anchor
> * Full Width Icon Left Button
> * Full Width Icon Right Button
> * Big Icon Left Anchor
> * Big Icon Right Anchor
> * Big Icon Left Button
> * Big Icon Right Button
> * Small Icon Left Anchor
> * Small Icon Right Anchor
> * Small Icon Left Button
> * Small Icon Right Button

Let's just add these to the big and small pages like we did for unstyled. We only need the primary icon color, as the icon page has all the other colors.

**Comment by andyvanavery31** [comment]
@karlikpj I updated the ticket so that it includes the "Icon Left Primary" and "Icon Right Primary" on the Big, Small, and Full Width Pages (both anchor and button). Please add those cases to those pages, and while you're making changes, can you please change the folder name "FullWidth" to "Full Width" (in the Modifiers folder), and change the "TestCases" folder name to "Test Cases".

Cc: @bryanpizzillo @olitharp-nci @sarinapadilla

**Review by andyvanavery31 — CHANGES_REQUESTED** [review]
`ncids_usa-button_nci-full-width_anchor_0_document_0_desktop.png`,  `ncids_usa-button_nci-full-width_button_0_document_0_desktop.png`, and `ncids_usa-button_unsuported_outline_color_0_document_0_mobile-lg.png` are giving a "Couldn't find story matching..." error. Could you please fix them?

**Comment by andyvanavery31** [comment]
It looks like `ncids_usa-button_unsuported_outline_color_0_document_0_mobile-lg.png`  got changed to `ncids_usa-button_testcases_unsuported_outline_color_0_document_0_mobile-lg.png` and is now throwing the same error. Could you check it? @karlikpj

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

File: testing/ncids-css-testing/stories/components/usa-button/usa-button.scss [hunk 1]
```diff
@@ -2,12 +2,21 @@
...
 	margin-bottom: 1rem;
 
 	&.usa-button--unstyled {
 		margin-left: 1rem;
 		margin-right: 1rem;
-
 	}
 }
+
+.section-header {
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/usa-button/usa-button.scss:14]: let's chat

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
Full width buttons with icons does missing disabled state

**Comment by olitharp-nci** [comment]
# Button component known issues

1. **Icon buttons do not emulate USWDS Button Icons.** Button icons need to be corrected to match USWDS as per the initial request. Additional requirements needed by design should possibly be an NCI variant. Consider creating button icons as a separate component to align with other design systems.
2. **Button documentation page needs a revamp.** We had a meeting in the past about issues with the documentation page missing, incorrect, or confusing information, and now it is worse.
   1.  No documentation on external links
   2. No documentation on buttons with icons
   3. Code snippets only exist for the default and secondary buttons
   5. We should not be highlighting `usa-button--hover` `usa-button--active` and `usa-focus` in code snippets. These classes should be visual only.
   6. Variations are wrong
   7. Modifications are wrong
   
8. **The NCI variant small button should not have left and right padding.** See storybook example where the small button is the only in the list to have extra padding. The button should function exactly as all other like buttons.

**Comment by bennettcc** [comment]
Per 4/30 standup - known issues #1-3 will be created into another story. Item #4 will be created into another story.

All are nice to haves for NCIDS launch (can sit at the bottom of the NCIDS Launch backlog

@andyvanavery31

**Review by andyvanavery31 — CHANGES_REQUESTED** [review]
Can you please update all the regression images so that there are two images per story:
1. Mobile breakpoint
2. Tablet Lg breakpoint

These will cover the button going full width at mobile and will show them in their "normal" view without wrapping.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
One issue in a package.json and also the following:

Full width icon left looks broken:
image
https://designsystem-dev.cancer.gov/pr-1614/storybook/?path=/story/components-button-modifiers-icon-icon-left--anchor

https://designsystem-dev.cancer.gov/pr-1614/storybook/?path=/story/components-button-modifiers-icon-icon-left--button

Also shouldn't this have a title with "unstyled" in it?

File: testing/ncids-css-testing/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "ncids-css-testing",
-	"version": "2.7.0",
+	"version": "2.6.0",
```
[the remark this answers is not in the document]
Comment (olitharp-nci) [reply testing/ncids-css-testing/package.json]: tis

**Review by bryanpizzillo — APPROVED** [review]
Looks good

**Review by andyvanavery31 — APPROVED** [review]
Looks good to me!

**Review by arcepaul — APPROVED** [review]

**Pull request merged**